services/web/src/screens/Invites/Actions.js

```diff
@@ -0,0 +1,83 @@
+import { ActionIcon, Menu, Text } from '@mantine/core';
+import { modals } from '@mantine/modals';
+import { showNotification } from '@mantine/notifications';
+
+import { IconDotsVertical, IconTrash, IconRepeat } from '@tabler/icons-react';
+
+import { request, useRequest } from 'utils/api';
+import ConfirmModal from 'components/ConfirmModal';
+import { notifications } from '@mantine/notifications';
+
+export default function InviteActions({ invite, reload }) {
+  const resentRequest = useRequest({
+    method: 'POST',
+    path: `/1/invites/${invite.id}/resend`,
+    manual: true,
+    onSuccess: () => {
+      showNotification({
+        position: 'top-right',
+        title: 'Invite re-sent',
+        color: 'green',
+      });
+    },
+    onError: (error) => {
+      notifications.show({
+        position: 'top-right',
+        title: 'Failed to re-send invite',
+        message: error.message,
+        color: 'red',
+      });
+    },
+  });
+
+  function openDeleteModel() {
+    modals.open({
```

Comment:
I see a lot of this: `modals.open`. Is this a mantine thing? My immediate reaction is that you lose a lot of the good parts of React being declarative when you do things in this imperative way.

Reply:
We talked about it sucks, but it seems to be the mantine way. I do want to revisit it at somepoint.